examples/go-counter/main.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/marigold-dev/deku/sdks/deku_go_interop"
+)
+
+type message struct {
+	Action string
+}
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func log(message string) {
+	colored := fmt.Sprintf("\x1b[%dm%s\x1b[0m", 31, message)
+	fmt.Fprintln(os.Stderr, colored)
+}
+
+func main() {
+	state_transition := func(sender_address string, tx_hash string, input []byte) (return_err error) {
+		var message message
+		log(fmt.Sprintf(
+				"State transition received. Sender: %s, Tx hash: %s, Payload: %s,",
+				sender_address,
+				tx_hash,
+				string(input)))
+		err := json.Unmarshal(input, &message)
+		check(err)
+		counter_bytes := deku_go_interop.Get("counter")
+		var counter *int
+		json.Unmarshal(counter_bytes, &counter)
+		var new_counter int
+		switch message.Action {
+		case "Increment":
```

Comment:
Why pattern-matching on a string? Isn't it possible to use a sum-type? _(I don't know go, this is a naive question)_

Reply:
Go does not have easy sum types.